### PR TITLE
Vector

### PR DESCRIPTION
**Pull request by Vtek**
### Issue
Vector : #3 

### Information
- Basic Vector 2d implementation + Unit tests
- Code documentation
- TSLint rules

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (Gilles-Lengy) [inline tslint.json]: This TSLint file shouldn't be part of this pull request.

Comment (Vtek) [inline tslint.json]: Ok, no problem i agree with you on this point !
I am doing the change right now :)
